src/components/OcAccordionItem.spec.js

```diff
@@ -0,0 +1,91 @@
+import { config, shallowMount, mount } from '@vue/test-utils';
+import OcAccordionItem from './OcAccordionItem.vue';
+config.showDeprecationWarnings = false
+
+describe('OcAccordionItem', () => {
+  const defaultHeadingLevel = 3
+  const mockMethods = {
+    toggleExpanded: jest.fn()
+  }
+  function getWrapperWithProps(props) {
+    return shallowMount(OcAccordionItem, {
+      propsData: {
+        title: 'Test title',
+        ...props
+      }
+    })
+  }
+  function getWrapperWithMockMethods() {
+    return mount(OcAccordionItem, {
+      propsData: {
+        title: 'Test title',
+      },
+      methods: mockMethods
+    })
+  }
+  describe('default behaviours', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it('should set item id as default', () => {
+      expect(wrapper.attributes('id')).toContain('oc-accordion-id-')
+    })
+    it('should render title as heading with default heading level', () => {
+      const titleElement = wrapper.find(`h${defaultHeadingLevel}`)
+      expect(titleElement.exists()).toBeTruthy()
+      expect(titleElement.attributes('class')).toBe('oc-accordion-title')
+      expect(titleElement.find('.uk-width-expand').text()).toBe('Test title')
+    })
+  })
+  it('should render description if given', () => {
+    const wrapper = getWrapperWithProps({
+      description: 'Test description'
+    })
+    const descriptionElement = wrapper.find('.oc-text-muted')
+    expect(descriptionElement.exists()).toBeTruthy()
+    expect(descriptionElement.text()).toBe('Test description')
+  })
+  describe('icon prop', () => {
+    describe('when icon is not given', ()=> {
+      let wrapper
+      wrapper = getWrapperWithProps({})
+      it('should not render any icon', () => {
+        const titleHeadingSpan = wrapper.get('span.uk-width-1-1')
+        expect(titleHeadingSpan.findAll('.oc-mr-s').length).toBe(0)
+        expect(titleHeadingSpan).toMatchSnapshot()
+      })
+    })
+    describe('when icon is given', ()=> {
+      it('should render an icon on the left side of title', () => {
+        const wrapper = getWrapperWithProps({
+          icon: 'mdi-icon'
+        })
+        const titleHeadingSpan = wrapper.get('span.uk-width-1-1')
+        expect(titleHeadingSpan.findAll('.oc-mr-s').length).toBe(1)
+        expect(titleHeadingSpan).toMatchSnapshot()
+      })
+      it ('should render extra space before description if given', () => {
```

Comment:
Too much level of detail. We could later on introduce visual regression tests for things like spacing, but please omit such things in unit tests. Unit tests should be about functionality, not so much about appearance.

Reply:
fixed